src/herder/HerderImpl.cpp

```diff
@@ -450,7 +459,40 @@ TransactionQueue::AddResult
 HerderImpl::recvTransaction(TransactionFrameBasePtr tx, bool submittedFromSelf)
 {
     ZoneScoped;
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    TransactionQueue::AddResult result;
+
+    // Allow txs of the same kind to reach the tx queue in case it can be
```

Comment:
I'm not sure I understand this logic, do we actually replace txs with the same seq num? Even if we do, I'm not sure if it's correct to only allow one certain kind of transactions...

Reply:
discussed offline, and agreed on the correctness of this logic. Herder cross checks the two queues, but still allows txs of the same kind to enter the respective queue in case it's a fee bump. If the tx can't actually replace an older tx, then the queue will return an appropriate error.